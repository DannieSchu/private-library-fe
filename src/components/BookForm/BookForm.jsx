import React, { useState } from 'react';
import { postAuthor, postBook } from '../../services/bookAPI';
import { useDispatch } from 'react-redux';
import { addBook } from '../../actions/booksActions';

const BookForm = () => {
  const dispatch = useDispatch();

  const [title, setTitle] = useState('');
  const [author, setAuthor] = useState('');
  const [pages, setPages] = useState(null);
  const [genre, setGenre] = useState('literature');

  const handleSubmit = event => {
    event.preventDefault();

    postAuthor(author)
      .then(authorId => {
        postBook({ authorId, title, genre, pages })
          .then(book => {
            dispatch(addBook(book));
          });
      });
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={author} onChange={({ target }) => setAuthor(target.value)} placeholder="Author" />
      <input type="text" value={title} onChange={({ target }) => setTitle(target.value)} placeholder="Title" />
      <label htmlFor="genre">Genre:</label>
      <select id="genre" onChange={({ target }) => setGenre(target.value)}>
        <option value="literature">Literature</option>
        <option value="science fiction">Science Fiction</option>
        <option value="horror">Horror</option>
        <option value="satire">Satire</option>
      </select>
      <input type="number" min={0} value={pages} onChange={({ target }) => setPages(target.value)} placeholder="Page count" />
      <button>Add to Shelf</button>
    </form>
  );
};

export default BookForm;

