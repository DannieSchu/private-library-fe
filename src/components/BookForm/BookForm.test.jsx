import React from 'react';
import { shallow } from 'enzyme';
import BookForm from './BookForm';

describe('BookForm component', () => {
  it('renders BookForm', () => {
    const wrapper = shallow(<BookForm />);
    expect(wrapper).toMatchSnapshot();
  });
});

