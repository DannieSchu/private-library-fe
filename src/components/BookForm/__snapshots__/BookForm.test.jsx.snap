// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookForm component renders BookForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Author"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Title"
    type="text"
    value=""
  />
  <label
    htmlFor="genre"
  >
    Genre:
  </label>
  <select
    id="genre"
    onChange={[Function]}
  >
    <option
      value="literature"
    >
      Literature
    </option>
    <option
      value="science fiction"
    >
      Science Fiction
    </option>
    <option
      value="horror"
    >
      Horror
    </option>
    <option
      value="satire"
    >
      Satire
    </option>
  </select>
  <input
    min={0}
    onChange={[Function]}
    placeholder="Page count"
    type="number"
    value={null}
  />
  <button>
    Add to Shelf
  </button>
</form>
`;
